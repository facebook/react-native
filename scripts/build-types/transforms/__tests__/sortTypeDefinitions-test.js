/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

const sortTypeDefinitionsVisitor = require('../sortTypeDefinitions.js');
const babel = require('@babel/core');
const {promises: fs} = require('fs');
const path = require('path');

async function translate(code: string): Promise<string> {
  const result = await babel.transformAsync(code, {
    plugins: ['@babel/plugin-syntax-typescript', sortTypeDefinitionsVisitor],
  });

  return result.code;
}

describe('sortTypeDefinitions', () => {
  test('should divide top-lelvel declarations into exported and non-exported sections', async () => {
    const code = await fs.readFile(
      path.join(__dirname, '../__fixtures__/sortTypeDefinitions.d.ts'),
      'utf-8',
    );
    const result = await translate(code);
    expect(result).toMatchSnapshot();
  });
});
