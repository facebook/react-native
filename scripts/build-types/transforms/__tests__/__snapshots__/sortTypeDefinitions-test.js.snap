// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortTypeDefinitions should divide top-lelvel declarations into exported and non-exported sections 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as React from \\"react\\";
export declare type ExportedFoo = \\"A\\" | \\"B\\";
export declare type ExportedBar = null | string | number;
export declare class ExportedBaz {
  foo: string;
}
export declare function exportedFn(): void;
declare type Foo = \\"A\\" | \\"B\\";
declare type Bar = null | string | number;
declare class Baz {
  foo: string;
}
declare function fn(): void;"
`;
